src/js/controllers/modals/inputAmount.js

```diff
@@ -0,0 +1,180 @@
+'use strict';
+
+angular.module('copayApp.controllers').controller('inputAmountController', function($rootScope, $scope, $filter, $timeout, $ionicScrollDelegate, profileService, platformInfo, lodash, configService, go, rateService) {
+  var unitToSatoshi;
+  var satToUnit;
+  var unitDecimals;
+  var self = $scope.self;
+
+  $scope.init = function() {
+    var config = configService.getSync().wallet.settings;
+    $scope.unitName = config.unitName;
+    $scope.alternativeIsoCode = config.alternativeIsoCode;
+    $scope.specificAmount = $scope.specificAlternativeAmount = '';
+    $scope.isCordova = platformInfo.isCordova;
+    unitToSatoshi = config.unitToSatoshi;
+    satToUnit = 1 / unitToSatoshi;
+    unitDecimals = config.unitDecimals;
+    $scope.resetAmount();
+    $timeout(function() {
+      $ionicScrollDelegate.resize();
+    }, 100);
+  };
+
+  $scope.shareAddress = function(uri) {
+    if ($scope.isCordova) {
+      window.plugins.socialsharing.share(uri, null, null, null);
+    }
+  };
+
+  $scope.toggleAlternative = function() {
+    $scope.showAlternativeAmount = !$scope.showAlternativeAmount;
+
+    if ($scope.amount && isExpression($scope.amount)) {
+      var amount = evaluate(format($scope.amount));
+      if ($scope.showAlternativeAmount)
+        $scope.globalResult = '= ' + $filter('formatFiatAmount')(amount);
+      else
+        $scope.globalResult = '= ' + profileService.formatAmount(amount * unitToSatoshi, true);
+    }
+  };
+
+  function checkFontSize() {
+    if ($scope.amount && $scope.amount.length >= 13) $scope.smallFont = true;
+    else $scope.smallFont = false;
+  };
+
+  $scope.pushDigit = function(digit) {
+    checkFontSize();
+    if ($scope.amount && $scope.amount.length >= 19) return;
+
+    $scope.amount = ($scope.amount + digit).replace('..', '.');
+    processAmount($scope.amount);
+  };
+
+  $scope.pushOperator = function(operator) {
+    if (!$scope.amount || $scope.amount.length == 0) return;
+    $scope.amount = _pushOperator($scope.amount);
+
+    function _pushOperator(val) {
+      if (!isOperator(lodash.last(val))) {
+        return val + operator;
+      } else {
+        return val.slice(0, -1) + operator;
+      }
+    };
+  };
+
+  function isOperator(val) {
+    var regex = /[\/\-\+\x\*]/;
+    var match = regex.exec(val);
+    if (match) return true;
+    return false;
+  };
+
+  function isExpression(val) {
+    var regex = /^\.?\d+(\.?\d+)?([\/\-\+\*x]\d?\.?\d+)+$/;
+    var match = regex.exec(val);
+
+    if (match) return true;
+    return false;
+  };
+
+  $scope.removeDigit = function() {
+    if ($scope.amount.toString().length == 1) {
```

Comment:
Does this test need to be here? It seems just slicing the last value and calling `processAmount()` should achieve the same result.